app/models/promo_campaign.rb

```diff
@@ -0,0 +1,4 @@
+class PromoCampaign < ApplicationRecord
+  has_many :promo_registrations
+  validates :name, uniqueness: true
```

Comment:
Would you add another validation that makes sure you clear out whitespace from the front and back and maybe middle? In your controller you're passing directly what admins are typing.

Reply:
Yes